Examples/Io/Root/src/RootTrackStatesWriter.cpp

```diff
@@ -450,175 +441,133 @@ ActsExamples::ProcessCode ActsExamples::RootTrackStatesWriter::writeT(
               state.smoothed() + K * (m - H * state.smoothed());
           auto unbiasedParamsCov =
               state.smoothedCovariance() - K * H * state.smoothedCovariance();
-          trackParamsOpt = std::make_pair(unbiasedParamsVec, unbiasedParamsCov);
+          return std::make_pair(unbiasedParamsVec, unbiasedParamsCov);
         }
+        return std::nullopt;
       };
 
       // fill the fitted track parameters
       for (unsigned int ipar = 0; ipar < 4; ++ipar) {
```

Comment:
actually, might be worth defining an `enum` like:
```c++
enum ParameterType {
	ePredicted,
	eFiltered,
	eSmoothed,
	eUnbiased,
	eSize
};
```

This would make the code a lot clearer, especially in the `if` clauses...